### PR TITLE
Document & better-type the v1 notification API

### PR DESCRIPTION
**Pull request by swansontec**

### CODE REVIEW

**Comment by GoodluckH** [comment]
Is this one still open? I thought these commits have been rebased to another PR for replacing express with serverlet?

**Comment by swansontec** [comment]
> Is this one still open? I thought these commits have been rebased to another PR for replacing express with serverlet?

Both of these commits are brand new, and need to be reviewed. They are present in #32, but that is now a future PR based on this one. Edit: Oops, I forgot to do the future workflow correctly, it looks like, but that was the idea.

**Review by GoodluckH — APPROVED** [review]
LGTM

File: src/server/controllers/NotificationController.ts [hunk 1]
```diff
@@ -1,50 +1,46 @@
...
-    console.error(
-      `Failed to send notifications to user ${String(req.body.userId)} devices`,
-      err
-    )
-    res.status(500).json(err)
-  }
+const asSendNotificationBody = asObject({
+  title: asString,
+  body: asString,
+  data: asOptional(asMap(asUnknown)),
```
Comment (GoodluckH) [inline src/server/controllers/NotificationController.ts:44]: Should it be asObject?

Comment (swansontec) [reply src/server/controllers/NotificationController.ts]: You are right!

**Pull request merged**